internal/session/repository_session.go

```diff
@@ -306,28 +314,28 @@ func (r *Repository) TerminateSession(ctx context.Context, sessionId string, ses
 		func(reader db.Reader, w db.Writer) error {
 			rowsAffected, err := w.Exec(ctx, terminateSessionCte, []interface{}{sessionId, sessionVersion})
 			if err != nil {
-				return fmt.Errorf("unable to terminate session %s: %w", sessionId, err)
+				return errors.Wrap(err, op, errors.WithMsg(fmt.Sprintf("unable to terminate session %s", sessionId)))
 			}
 			if rowsAffected == 0 {
-				return fmt.Errorf("unable to terminate session %s", sessionId)
+				return errors.New(errors.SessionNotFound, op, fmt.Sprintf("unable to terminate session %s", sessionId))
```

Comment:
I made this session not found, but I am not convinced that is always the case for a non-errored failure to update?

Reply:
Would `InvalidSessionState` be a better err here?

Reply:
bad state or unknown error.